release_tools/release-utils.sh

```diff
@@ -0,0 +1,143 @@
+BUILDDIR="$CONTENT_REPO_ROOT/build"
+
+CMAKE_FILE="$CONTENT_REPO_ROOT/CMakeLists.txt"
+MAJOR_VERSION=$(awk '/SSG_MAJOR_VERSION /{print substr($2, 1, length($2)-1)}' $CMAKE_FILE)
+MINOR_VERSION=$(awk '/SSG_MINOR_VERSION /{print substr($2, 1, length($2)-1)}' $CMAKE_FILE)
+PATCH_VERSION=$(awk '/SSG_PATCH_VERSION /{print substr($2, 1, length($2)-1)}' $CMAKE_FILE)
+
+version=$MAJOR_VERSION.$MINOR_VERSION.$PATCH_VERSION
+let PATCH_VERSION++
+next_version=$MAJOR_VERSION.$MINOR_VERSION.$PATCH_VERSION
+
+die()
+{
+    echo "$1"
+    exit 1
+}
+
+check_jenkins_credentials()
+{
+    test -n "$JENKINS_USER" || die "We don't know your Jenkins user, so we can't proceed. It usually is the same as your GitHub user. Get to know your User ID in your profile page in Jenkins, and put it in the .env file, so it contains the line JENKINS_USER='<jenkins user id>'"
+    test -n "$JENKINS_TOKEN" || die "We don't know your Jenkins API token, so we can't proceed. Generate one on your Configure page in Jenkins and put it in the .env file, so it contains the line JENKINS_TOKEN='<jenkins token>'"
+}
+
+check_github_credentials()
+{
+    test -n "$GITHUB_TOKEN" || die "We don't know your GitHub token, so we can't proceed. Get one on https://github.com/settings/tokens and put it in the .env file, so it contains the line GITHUB_TOKEN='<github token>'"
+}
+
+check_for_clean_repo()
+{
+    echo Checking whether the repository is clean...
+    # check that there is nothing to report by 'status' except untracked files.
+    git status --porcelain=v2 | grep -q --invert-match '^?' && die "The repository is not clean, stash your changes to proceed."
+}
+
+check_release_is_ok()
+{
+    check_github_credentials
+    # check that release $version doesn't exist yet
+    python3 content_gh.py $OWNER $REPO $GITHUB_TOKEN $version check || die
+}
+
+check_jenkins_jobs()
+{
+    check_jenkins_credentials
+    echo Checking Jenkins Jobs
+    python3 jenkins_ci.py --jenkins-user $JENKINS_USER --jenkins-token $JENKINS_TOKEN check
+
+}
+
+check_rhel_stig_ids()
+{
+    echo Checking missing STIG IDS...
+    echo Building RHEL6 and RHEL7 content
+    ncpu=$(nproc) # This won't return number of physical core, but it shouldn't be problem
+    (cd "$BUILDDIR" && cmake .. && make -j $(ncpu) rhel7 rhel6) &> rhel_build.log || die "Error building RHEL7 content, check rhel_build.log file for errors"
+    (PYTHONPATH=.. python3 ../build-scripts/profile_tool.py stats -b ../build/ssg-rhel6-xccdf.xml --missing-stig-ids --profile stig) > rhel6-stig-ids.log
+    echo :: Check rhel6-stig-ids.log for rules missing STIG IDs
+    (PYTHONPATH=.. python3 ../build-scripts/profile_tool.py stats -b ../build/ssg-rhel7-xccdf.xml --missing-stig-ids --profile stig) > rhel7-stig-ids.log
+    echo :: Check rhel7-stig-ids.log for rules missing STIG IDs
+
+}
+
+make_dist()
+{
+    if [ -e artifacts/scap-security-guide-$version.tar.bz2 ]; then
+        echo Tarball already exists
```

Comment:
If I understand it well this mean there is a leftover file with unknown contents. Do you think it should terminate the script? Or remove the file?

Reply:
> If I understand it well this mean there is a leftover file with unknown contents.

Not necessarily. `make_dist()` is called by `build.sh`, which can be called multiple times while waiting for builds to finish. This is to check whether the tarball already exists, and avoid repeatedly generating it.

The tarball is removed when `cleanup.sh` is run, whole `artifacts/` directory is removed. See this [line](https://github.com/ComplianceAsCode/content/pull/4666/files#diff-184c7a1349c20ffd1f249962c52075a9R138).

> Do you think it should terminate the script? Or remove the file?

The `make_dist()` command can probably be moved to a different script.
I'll consider this when getting rid of the `.sh` scripts and everything is "pythonized".